betaflight/betaflight
# Voltage detection method

The isVoltageStable method before compared filtered and unfiltered voltage data to determine if the voltage is stable or not. This lead to multiple instances of wrong detected cell counts, thus profile switching based on cell count was screwed.

This PR changes it so, that the method always uses filtered data to check for stable voltage. Also, the previous value gets set in 1 second intervals. Also, I changed from 200mV to 30mV difference, because filtered values are way more stable.

To test this PR:
Set your quad up for BATTERY debug mode and observe Debug 4, which will go to 1 if the voltage is set stable. It should be some seconds later than before and at a state of really stable voltage.
